dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -370,7 +370,7 @@ public CloseableClient build() {
             SSLSocketFactory rawSocketFactory = conf.sslSocketFactory();
             SSLConnectionSocketFactory sslSocketFactory = new SSLConnectionSocketFactory(
                     MetricRegistries.instrument(conf.taggedMetricRegistry(), rawSocketFactory, name),
-                    new String[] {"TLSv1.2"},
+                    null,
```

Comment:
seems like setting `null` will result in all TLS protocols being enabled. What happens if we declare 1.3 but do not have the corresponding ciphers? i.e. could we be more explicit and set both 1.2 and 1.3?

Reply:
If we explicitly set both, we'll throw exceptions on java versions which do not support 1.3. Using null delegates to hc5 to support all available versions, and exclude the weak versions, which results in either `[TLSv1.2]` or `[TLSv1.3, TLSv1.2]`.

We would have to either get an instance of the SSLContext (which isn't provided by the ClientConfiguration API) or create a socket to access this information.

Reply:
ok cool makes sense, mind adding a comment for posterity?

Reply:
:+1: